functional/FunctionalTest.java

```diff
@@ -247,6 +252,64 @@ public static void putObject_test6() throws Exception {
   }
 
   /**
+   * Test: client side encryption: putObject(String bucketName, String objectName, InputStream stream, long size, String
+   * contentType, EncryptionMaterials encryptionMaterials).
+   */
+  public static void putObject_test7() throws Exception {
+
+    System.out.println(
+        "Test: client side encryption (AES 256 bit): putObject(String bucketName, String objectName, InputStream "
+            + "stream,long size, String contentType, SecureKey key).");
+    String fileName = createFile(13 * MB);
+    InputStream is = Files.newInputStream(Paths.get(fileName));
+
+    // Generate symmetric 256 bit AES key.
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(128);
+    SecretKey symKey = symKeyGenerator.generateKey();
+
+    try {
+      client.putObject(bucketName, fileName, is, 20 * 1024 * 1024, null, symKey);
+    } catch (InsufficientDataException e) {
```

Comment:
You should not pass excessive size value (pass 13 * MB) and handle `InsufficientDataException` for this test.